lib/tasks/data_key_validation.rake

```diff
@@ -0,0 +1,52 @@
+
+namespace :data_key_validation do
+  desc "Validates the value of ENV['CONJUR_DATA_KEY']"
+  # This task validate the master data key which is stored in `ENV['CONJUR_DATA_KEY']`
+  # Once the db is created, a secret will be created using the master key.
+  # Each time the server is restarted this task will attempt to retrieve the
+  # secret, if the provided key does not match the original master the secret retrieval
+  # will fail.
+  #
+  # What's the 'environment' task in Rake?
+  # the :validate => :environment means "do environment before validate".
+  # You can get access to your models, and in fact, your whole environment
+  # by making tasks dependent on the environment task.
+  task :validate => [:environment] do
+    data_key = ENV['CONJUR_DATA_KEY']
+
+    if data_key.nil? || data_key.empty?
+      raise Errors::System::EmptyConjurDataKey
+      exit 1
+    end
+
+    # try get the master key validator resource
+    resource_id = "!:!:data-key-validator-resource"
+
+    unless Resource[resource_id]
+      # create the master key validator resource
+      # and store a secret using the master key
+      role_id = "!:!:data-key-validator"
+
+      begin
+        Role.create(role_id: role_id)
+        Resource.create(resource_id: resource_id, owner_id: role_id)
+        Secret.create(resource_id: resource_id, value: "Don't Panic.")
```

Comment:
Are these resources visible to the end user? If so, I think we should find another way to test the validity of the data key to avoid introducing unexpected resources for the user.

Reply:
The key is used only in secrets and Host Tokens.

Reply:
i'd still like to get an answer to @rafis3 's question - are these resources visible to the end user? are they visible only to the admin user?

Reply:
also - what's `Don't Panic.`? can we have something more official like `data-key-validator-secret`?

Reply:
> also - what's `Don't Panic.`? can we have something more official like `data-key-validator-secret`?

It's a quote from Hitchhiker's guide to the galaxy. No one is suppose to read this value. Consider it an easter egg.

Reply:
> i'd still like to get an answer to @rafis3 's question - are these resources visible to the end user? are they visible to the admin user?

It will have to wait for the next boost.

Reply:
i prefer that we have `data-key-validator-secret` instead of a joke that not everyone will understand